py/dml/expr.py

```diff
@@ -292,6 +335,32 @@ def copy(self, site):
 
 mkNullConstant = NullConstant
 
+class Orphan(Expression):
+    """Expressions that evaluate to a value that is allocated on the stack, but
+    not belonging to any local variable. Archetypical example are function
+    applications."""
```

Comment:
This description was written in a better time when orphans were directly tied to C concepts. What they are nowadays are not so simple (`Expression.orphan` has the better description)